xtp/src/libxtp/dftengine/dftengine.cc

```diff
@@ -348,6 +355,296 @@ bool DFTEngine::Evaluate(Orbitals& orb) {
   return true;
 }
 
+bool DFTEngine::EvaluateActiveRegion(Orbitals& orb) {
+
+  // reading in the orbitals of the full DFT calculation
+  tools::EigenSystem embeddingMOs;
+  embeddingMOs.eigenvalues() = orb.MOs().eigenvalues();
+  embeddingMOs.eigenvectors() = orb.MOs().eigenvectors();
+
+  // constructing the full electron density matrix
+  const Eigen::MatrixXd FullDensityMatrix = orb.DensityMatrixGroundState();
+  XTP_LOG(Log::error, *pLog_) << std::flush;
+  XTP_LOG(Log::error, *pLog_)
+      << TimeStamp() << " Starting Projection based dft-in-dft embedding"
+      << std::flush;
+
+  // reading the localized orbitals and update the occupied orbitals in MOs
+  XTP_LOG(Log::error, *pLog_)
+      << TimeStamp() << " Passing localized orbitals as the initial guess"
+      << std::flush;
+  Eigen::MatrixXd LMOs = orb.getPMLocalizedOrbital();
+
+  embeddingMOs.eigenvectors().leftCols(orb.getNumberOfAlphaElectrons()) = LMOs;
+
+  // determine the active and inactive electron densities
+  std::vector<Index> activeatoms =
+      IndexParser().CreateIndexVector(active_atoms_as_string);
```

Comment:
I would do this the moment you get the string, because then if there is an error it is caught earlier.